File: LeetCode/1/Solution.py
### 
# https://leetcode.com/problems/two-sum
# vishal
# 01/30/2019
###

def twoSums(nums, target):
    hmap={}
    for i in range(len(nums)):
        complement = target - nums[i]
        try:
            if hmap[complement]>-1:
                return [hmap[complement],i]
        except:
            hmap[nums[i]]=i


if __name__=="__main__":
    size=int(input())
    l = list(map(int, input().split(' ',size)))
    sum = int(input())
    result = twoSums(l,sum)
    print(result)

File: LeetCode/1/Solution.java
// https://leetcode.com/problems/two-sum
// vishal
// 01/30/2019

import java.util.*;

class Solution{
    public static int[] twoSum(int[] nums, int target){
        HashMap<Integer, Integer> hmap = new HashMap<>();
        for(int i=0;i<nums.length;++i){
            int complement = target - nums[i];
            if(hmap.containsKey(complement))
                return new int[]{hmap.get(complement),i};
            hmap.put(nums[i], i);
        }        
        return new int[]{-1,-1};
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int size = sc.nextInt();
        int[] v = new int[size];
        for(int i=0;i<size;++i){
            
            v[i]=sc.nextInt();
        }
        int sum=sc.nextInt();
        int []result=twoSum(v,sum);
        for(int i:result)
            System.out.print(i+" ");
    }
}